src/Orleans/Providers/ILeaseProvider.cs

```diff
@@ -81,6 +107,23 @@ public class LeaseRequest
         /// Duration of the lease
         /// </summary>
         public TimeSpan Duration { get; set; }
```

Comment:
Offline discussion agree to remove [Immutable] from the class. It is a trade-off between usability and minor perf cost